### PR TITLE
Added support for environment to be defined in Webpack CLI's `--mode` flag

### LINKED ISSUE #17: Need two options for bundled artefact
The build task should be able to generate two different variations of the bundled artefact:

- one for dev/staging, which includes source maps
- one for prod, which does not include source maps

This is seperate from webpack-serve, which is purely used during local development.

### PR DESCRIPTION
**Pull request by keeganstreet**
Resolves #17

### CODE REVIEW

File: packages/webpack-config/src/index.js [hunk 1]
```diff
@@ -1,55 +1,76 @@
...
+	mode = argv.mode;
+} else if (process.env.WEBPACK_SERVE) {
+	mode = 'development';
+}
+
 dd();
-console.log('Webpack Config Deloitte');
+console.log('Deloitte Digital Webpack Config');
 console.log('Current working directory:', process.cwd());
+console.log('Mode:', mode);
```
Comment (jennasalau) [inline packages/webpack-config/src/index.js]: I like this, what do you think about only logging the directory and mode when the --verbose flag is present? Just an idea.

File: packages/webpack-config/src/index.js [hunk 2]
```diff
@@ -1,55 +1,76 @@
+const argv = require('yargs').argv;
 const autoprefixer = require('autoprefixer')();
 const CleanWebpackPlugin = require('clean-webpack-plugin');
 const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 const path = require('path');
 const webpack = require('webpack');
 const dd = require('./dd.js');
 
+let mode = 'production';
+if (argv.mode === 'development' || argv.mode === 'production') {
```
Comment (jennasalau) [inline packages/webpack-config/src/index.js]: We shouldn't rely on the mode flag to derive this. For instance using `.env` or running the command with `NODE_ENV=production webpack` are also valid.

The mode flag is used to preset the environment variables. So its the environment variables that we should use as the source of truth.

**[mode option development](https://webpack.js.org/concepts/mode/)**

>  Provides process.env.NODE_ENV with value development. 

So this line should be something similar to:

`let mode = process.env.NODE_ENV || 'development';`

Comment (keeganstreet) [reply packages/webpack-config/src/index.js]: Hey Jenna, I can understand that we should also support the `NODE_ENV` environment variable, but the problem is, setting Webpack's `mode` flag does not modify the `process.env.NODE_ENV` variable, so we need to check both. How about this logic -

If `NODE_ENV` env var is set, use this as the mode.
Else if the `mode` argument is set, use this as the mode.
Else if the `WEBPACK_SERVE` env var is set, set the mode to development.

File: packages/webpack-config/src/index.js (hunk 1 above)
Comment (keeganstreet) [reply packages/webpack-config/src/index.js]: Good idea. That's consistent with Webpack only showing `PublicPath: /dist/` when the `--verbose` flag is present.

File: packages/webpack-config/src/index.js (hunk 2 above)
Comment (jennasalau) [reply packages/webpack-config/src/index.js]: Oh really :(

Yeah if thats the case then I think I like your logic. It covers all bases.

**Comment by jennasalau** [comment]
Yeah this looks really solid now 👍

**Pull request merged**